File: mybatis/java/src/main/java/examples/mybatis/mapper/PersonMapper.java
package examples.mybatis.mapper;

import examples.mybatis.entity.Person;
import org.apache.ibatis.annotations.*;

@Mapper
public interface PersonMapper {


    @Update("create table if not exists person(id integer primary key, name text, age integer)")
    void createTable();

    @Select("select * from person where id = ${id}")
    Person find(@Param("id") Long id);

    @Select("select * from person where name = #{name}")
    Person findByName(@Param("name") String name);

    @Insert("insert into person (id, name, age) values (#{id}, #{name}, #{age})")
    void insert(Person person);

}

File: mybatis/kotlin/src/main/kotlin/examples/mybatis/mapper/PersonMapper.kt
package examples.mybatis.mapper

import examples.mybatis.entity.Person
import org.apache.ibatis.annotations.*

@Mapper
interface PersonMapper {
    @Update("create table if not exists person(id integer primary key, name text, age integer)")
    fun createTable()

    @Select("select * from person where id = \${id}")
    fun find(@Param("id") id: Long?): Person?

    @Select("select * from person where name = #{name}")
    fun findByName(@Param("name") name: String?): Person?

    @Insert("insert into person (id, name, age) values (#{id}, #{name}, #{age})")
    fun insert(person: Person?)
}
